File: TextosJapones/src/main/java/org/jisho/textosJapones/model/services/EstatisticaServices.java
package org.jisho.textosJapones.model.services;

import org.jisho.textosJapones.controller.EstatisticaController.Tabela;
import org.jisho.textosJapones.database.dao.DaoFactory;
import org.jisho.textosJapones.database.dao.EstatisticaDao;
import org.jisho.textosJapones.model.entities.Estatistica;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

import java.util.List;

public class EstatisticaServices {

	private EstatisticaDao estatisticaDao = DaoFactory.createEstatisticaDao();

	public List<Estatistica> selectAll() throws ExcessaoBd {
		return estatisticaDao.selectAll();
	}

	public Estatistica select(String kanji, String leitura) throws ExcessaoBd {
		return estatisticaDao.select(kanji, leitura);
	}

	public List<Estatistica> select(String kanji) throws ExcessaoBd {
		return estatisticaDao.select(kanji);
	}

	public List<Tabela> pesquisa(String pesquisa) throws ExcessaoBd {
		return estatisticaDao.pesquisa(pesquisa);
	}

	public void insert(Estatistica obj) throws ExcessaoBd {
		if (!obj.getKanji().isEmpty())
			estatisticaDao.insert(obj);
	}

	public void insert(List<Estatistica> lista) throws ExcessaoBd {
		for (Estatistica obj : lista)
			if (!obj.getKanji().isEmpty())
				estatisticaDao.insert(obj);
	}

	public void update(Estatistica obj) throws ExcessaoBd {
		estatisticaDao.update(obj);
	}

	public void delete(Estatistica obj) throws ExcessaoBd {
		estatisticaDao.delete(obj);
	}

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/service/EstatisticaServices.kt
package br.com.fenix.processatexto.service

import br.com.fenix.processatexto.database.DaoFactory
import br.com.fenix.processatexto.database.dao.EstatisticaDao
import br.com.fenix.processatexto.model.entities.processatexto.japones.Estatistica
import br.com.fenix.processatexto.controller.EstatisticaController.Tabela
import java.sql.SQLException
import java.util.*


class EstatisticaServices {

    private val dao: EstatisticaDao = DaoFactory.createEstatisticaDao()

    @Throws(SQLException::class)
    fun selectAll(): List<Estatistica> = dao.selectAll()

    @Throws(SQLException::class)
    fun select(kanji: String, leitura: String): Optional<Estatistica> = dao.select(kanji, leitura)

    @Throws(SQLException::class)
    fun select(kanji: String): MutableList<Estatistica> = dao.select(kanji)

    @Throws(SQLException::class)
    fun pesquisa(pesquisa: String): List<Tabela> = dao.pesquisa(pesquisa)

    @Throws(SQLException::class)
    fun insert(obj: Estatistica) {
        if (obj.kanji.isNotEmpty()) {
            if (obj.getId() == null)
                obj.setId(UUID.randomUUID())
            dao.insert(obj)
        }
    }

    @Throws(SQLException::class)
    fun insert(lista: List<Estatistica>) {
        for (obj in lista)
            insert(obj)
    }

    @Throws(SQLException::class)
    fun update(obj: Estatistica) = dao.update(obj)

    @Throws(SQLException::class)
    fun delete(obj: Estatistica) = dao.delete(obj)

}